File: Level 1/성격 유형 검사하기/solution.py
def solution(surveys, choices):
    choice_map = {"R": 0, "T": 0, "C": 0, "F": 0, "J": 0, "M": 0, "A": 0, "N": 0}

    for survey, choice in zip(surveys, choices):
        if choice < 4:
            choice_map[survey[0]] += abs(choice - 4)
        elif choice > 4:
            choice_map[survey[1]] += abs(choice - 4)

    answer = ""

    for personality in ["RT", "CF", "JM", "AN"]:
        if choice_map[personality[0]] > choice_map[personality[1]]:
            answer += personality[0]
        elif choice_map[personality[0]] < choice_map[personality[1]]:
            answer += personality[1]
        else:
            answer += sorted(personality)[0]

    return answer

File: Level 1/성격 유형 검사하기/solution.js
function solution(surveys, choices) {
  const n = surveys.length;
  const choiceMap = {};

  surveys.forEach((survey) => {
    choiceMap[survey[0]] = 0;
    choiceMap[survey[1]] = 0;
  });

  for (let i = 0; i < n; i++) {
    const survey = surveys[i];
    const choice = choices[i];

    if (choice < 4) {
      choiceMap[survey[0]] = choiceMap[survey[0]] + Math.abs(choice - 4);
    } else if (choice > 4) {
      choiceMap[survey[1]] = choiceMap[survey[1]] + Math.abs(choice - 4);
    }
  }

  return ["RT", "CF", "JM", "AN"]
    .map((personality) =>
      choiceMap[personality[0]] > choiceMap[personality[1]]
        ? personality[0]
        : choiceMap[personality[0]] < choiceMap[personality[1]]
        ? personality[1]
        : [...personality].sort()[0]
    )
    .join("");
}
